docs/src/proposals/fee_transaction_priority.md

```diff
@@ -0,0 +1,210 @@
+---
+title: Fee Transaction Priority
+---
+
+Additional fees were introduced to transactions as a method to allow users to bid for priority for
+their transactions in the leader's queue.
+
+The fee priority of a transaction `T` can then be defined as `F(T)`, where `F(T)` is the "fee-per
+compute-unit", calculated by:
+
+`(additional_fee + base_fee) / requested_compute_units`
+
+To ensure users get fair priority based on their fee, the proposed scheduler for the leader must
+guarantee that given `T1` and `T2` in the pending queue, and `F(T1) > F(T2)`:
+
+1. `T1` should be considered for processing before `T2`
+2. If `T1` cannot be processed before `T2` because there's already a transaction currently being
+processed that contends on an account `A`, then `T2` should not be scheduled if it would grab
+any account locks needed by `T1`. This prevents lower fee transactions like `T2` from starving
+higher paying transactions like `T1`.
```

Comment:
To clarify, the starvation is:
- is a pre-existing issue due to banking-stage threads are isolated from each other, a TX that needs many accounts in one thread can be starved due to other threads keep submitting other TXs that take one of those accounts.
- it becomes necessary to be solved now because we are to promise prioritize Txs by fee/CU.
- to solve it, needs some kind central scheduling schema across banking-stage threads.

Reply:
if you have the following 5 txs:
1.  fee rate 300. write locks accounts: A, B, C
2. fee rate 200: write locks accounts: B, C, D
3. fee rate 250: write locks accounts: C, D
4. fee rate 400. write locks accounts: E, F
5. fee rate 500. write locks accounts: D

would the tx batching then be: [[5, 4, 1], [3], [2]]?

Reply:
1. 500: [A, B, C(read)]
2. 450: [A, C(write)]
3. 400: [C(read), D]

what about these transactions with account and rw flags? you could build a batches like:
- [[1,3], [2]]
- [[1], [2], [3]]

the second option would respect the fee ordering assuming you care about read lock fees, the first one would result in less batches